fondant/dataset.py

```diff
@@ -76,60 +105,124 @@ def load_dataframe(self, spec: FondantComponentSpec) -> dd.DataFrame:
 
         return df
 
-    def _upload_index(self, df: dd.DataFrame):
-        # get remote path
-        remote_path = self.manifest.index.location
-
-        # upload to the cloud
-        dd.to_parquet(
-            df,
-            remote_path,
-            schema=self.index_schema,
-            overwrite=True,
+    @staticmethod
+    def _create_write_dataframe_task(
+        *, df: dd.DataFrame, remote_path: str, schema: t.Dict[str, str]
+    ) -> dd.core.Scalar:
+        """
+        Creates a delayed Dask task to upload the given DataFrame to the remote storage location
+         specified in the manifest.
+        Args:
+            df: The DataFrame to be uploaded.
+            remote_path: the location to upload the subset to
+            schema: the schema of the dataframe to write
+        Returns:
+             A delayed Dask task that uploads the DataFrame to the remote storage location when
+              executed.
+        """
+
+        # Define task to upload index to remote storage
+        upload_index_task = dd.to_parquet(
+            df, remote_path, schema=schema, overwrite=True, compute=False
```

Comment:
Should we set `overwrite=True`? I think we should assume that the directory we are writing to is empty and otherwise fail?

Reply:
yeah indeed, this was easier for testing purposes but I think it can be removed